gui/src/app/pages/HomePage/AnalysisPyWindow/AnalysisPyWindow.tsx

```diff
@@ -0,0 +1,204 @@
+      />
+      <RightPane width={0} height={0} onImageOutputDiv={setImageOutputDiv} />
+    </Splitter>
+  );
+};
+
+type LeftPaneProps = {
+  width: number;
+  height: number;
+  imageOutputDiv: HTMLDivElement | null;
+  stanSampler: StanSampler | null;
+};
+
+const LeftPane: FunctionComponent<LeftPaneProps> = ({
+  width,
+  height,
+  imageOutputDiv,
+  stanSampler,
+}) => {
+  const [consoleOutputDiv, setConsoleOutputDiv] =
+    useState<HTMLDivElement | null>(null);
+  const { data, update } = useContext(ProjectContext);
+  const { draws, paramNames, numChains } = useSamplerOutput(
+    stanSampler || undefined,
+  );
+  const { globalData, scriptHeader } = useMemo(() => {
+    if (draws) {
+      return {
+        globalData: {
+          _sp_sampling: {
+            draws: transpose(draws),
+            paramNames,
+            numChains,
+          },
+        },
+        scriptHeader: getScriptHeaderForNonemptyDraws(),
+      };
+    } else {
+      return {
+        globalData: {},
+        scriptHeader: getScriptHeaderForEmptyDraws(),
+      };
+    }
+  }, [draws, paramNames, numChains]);
+  return (
+    <Splitter
+      direction="vertical"
+      width={width}
+      height={height}
+      initialPosition={(3 * height) / 5}
+    >
+      <AnalysisPyFileEditor
+        width={0}
+        height={0}
+        fileName="analysis.py"
+        fileContent={data.analysisPyFileContent}
+        editedFileContent={data.ephemera.analysisPyFileContent}
+        setEditedFileContent={(content) => {
+          update({
+            type: "editFile",
+            content,
+            filename: ProjectKnownFiles.ANALYSISPYFILE,
+          });
+        }}
+        onSaveContent={() => {
+          update({
+            type: "commitFile",
+            filename: ProjectKnownFiles.ANALYSISPYFILE,
+          });
+        }}
+        consoleOutputDiv={consoleOutputDiv}
+        imageOutputDiv={imageOutputDiv}
+        readOnly={false}
+        globalData={globalData}
+        scriptHeader={scriptHeader}
+      />
+      <ConsoleOutputWindow
+        width={0}
+        height={0}
+        onDivElement={setConsoleOutputDiv}
+      />
+    </Splitter>
+  );
+};
+
+type ConsoleOutputWindowProps = {
+  width: number;
+  height: number;
+  onDivElement: (div: HTMLDivElement) => void;
+};
+
+const ConsoleOutputWindow: FunctionComponent<ConsoleOutputWindowProps> = ({
+  width,
+  height,
+  onDivElement,
+}) => {
+  return (
+    <div
+      style={{ position: "absolute", width, height, overflowY: "auto" }}
+      ref={onDivElement}
+    />
+  );
+};
+
+type ImageOutputWindowProps = {
+  width: number;
+  height: number;
+  onDivElement: (div: HTMLDivElement) => void;
+};
+
+const ImageOutputWindow: FunctionComponent<ImageOutputWindowProps> = ({
+  width,
+  height,
+  onDivElement,
+}) => {
+  return (
+    <div
+      style={{ position: "absolute", width, height, overflowY: "auto" }}
+      ref={onDivElement}
+    />
+  );
+};
```

Comment:
These two components are currently identical--are they expected to diverge in the future?

Reply:
Yeah I think it's likely they will since the type of content is very different between them. Right now we construct the elements that go in these using non-react DOM-manipulation, but in the future we might be able to figure out how to set the content as props/state somehow.